### PR TITLE
Fix #3155: Allow toggling on leaderboard when submission is submitted/runnning

### LINKED ISSUE #3155: [FrontendV2] Allow toggling Show on leaderboard when submission is submitted/runnning
### Description

When a participant goes to `My Submission` page and select a submission they can't toggle `Show on leaderboard` checkbox when status of submission is `Submitted/Running`. We should allow toggling of this checkbox even when submission is not in `Finished` state.

Steps to reproduce:

1. Select a challenge
2. Go to Submit tab and make a submission
3. Go to My Submissions tab
4. Select a submission with `Submitted/Running` status and try to toggle `Show on Leaderboard` checkbox.

Here's a screenshot for reference:

Screen Shot 2021-01-20 at 9 17 55 PM

### PR DESCRIPTION
**Pull request by burnerlee**

### CODE REVIEW

File: frontend_v2/src/app/components/challenge/challengesubmissions/challengesubmissions.component.html [hunk 1]
```diff
@@ -166,7 +166,13 @@ <h5 class="fw-light">My Submissions</h5>
                 <div class="element">
                   <mat-checkbox
                     [checked]="element.is_public"
-                    [disabled]="element.status !== 'finished' || !selectedPhase['leaderboard_public']"
+                    [disabled]="
+                      !(
+                        element.status === 'finished' ||
```
Comment (Ram81) [inline frontend_v2/src/app/components/challenge/challengesubmissions/challengesubmissions.component.html:171]: @burnerlee the `Show on Leaderboard` option should be disabled only when submission status is either `cancelled` or `failed`

**Review by Ram81 — CHANGES_REQUESTED** [review]

**Comment by Ram81** [comment]
@burnerlee can you share a gif/demo showing toggling is working when status of submission is `submitted` or `running` or `submitting`?

**Comment by RishabhJain2018** [comment]
@burnerlee Any updates on this?

**Comment by burnerlee** [comment]
Yeah I'll just upload this, the latest master is giving some issues in running, I'll open an issue for this

**Comment by burnerlee** [comment]
@Ram81

**Review by Ram81 — APPROVED** [review]
LGTM 👍

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**